ocean_lib/enforce_typing_shim.py

```diff
@@ -15,6 +16,15 @@ def setup_enforce_typing_shim():
 
 
 def enforce_types_shim(func):
-    if not ConfigProvider.get_config()["util"].getboolean("typecheck"):
+    try:
+        c = ConfigProvider.get_config()
+    except AssertionError:  # handle if ConfigProvider.set_config() not done yet
```

Comment:
can this AssertionError mean something else? Maybe it can be thrown for "incorrect" values as well as "missing" ones. When do we want to override with the defaults? Should we warn the end user in some way?

Reply:
Right now config_provider.ConfigProvider has the following. 
```python
    def get_config():
        """Get a Config instance."""
        if not ConfigProvider._config:
            raise AssertionError("set_config first.")
        return ConfigProvider._config
```

We could make a more specific label than `AssertionError` or get fancier in how to handle it. However soon I will work on #202 "singletons for Web3Provider and more is surprising, non-standard, and won't work for multi-chain". In that, ConfigProvider is gonna get revamped anyway, and almost certainly change the function above. 

Therefore for _this_ issue I just want to do something surgical and move on. How does that sound?

Reply:
Great! I will approve and let's move on. If it's documented in other issues, that's all I need to hear 👍